### PR TITLE
Issue 395 blast search backend implementation canceled status

### PR DESCRIPTION
**Pull request by okolesn**
# Description
Cancel task improvement for 
[[BLAST Search] Backend implementation](https://github.com/epam/NGB/issues/395).

Task status will be updated even If task wasn't cancelled. Task status is taken from error message.

### CODE REVIEW

File: server/catgenome/src/main/java/com/epam/catgenome/entity/blast/BlastTaskStatus.java [hunk 1]
```diff
@@ -36,7 +36,7 @@
 
     private long id;
```
Comment (SilinPavel) [inline server/catgenome/src/main/java/com/epam/catgenome/entity/blast/BlastTaskStatus.java]: could we make it `int` and don't cast it for `idMap`

Comment (SilinPavel) [reply server/catgenome/src/main/java/com/epam/catgenome/entity/blast/BlastTaskStatus.java]: also could be final

File: server/catgenome/src/main/java/com/epam/catgenome/entity/blast/BlastTaskStatus.java [hunk 2]
```diff
@@ -36,7 +36,7 @@
...
     private boolean finalStatus;
```
Comment (SilinPavel) [inline server/catgenome/src/main/java/com/epam/catgenome/entity/blast/BlastTaskStatus.java]: should be final

File: server/catgenome/src/main/java/com/epam/catgenome/entity/blast/BlastTaskStatus.java [hunk 3]
```diff
@@ -36,7 +36,7 @@
...
-    private static Map<Long, TaskStatus> idMap = new HashMap<>((int) CREATED.getId());
+    private static Map<Long, BlastTaskStatus> idMap = new HashMap<>((int) CREATED.getId());
```
Comment (SilinPavel) [inline server/catgenome/src/main/java/com/epam/catgenome/entity/blast/BlastTaskStatus.java]: could be final

**Review by SilinPavel — CHANGES_REQUESTED** [review]

File: server/catgenome/src/main/java/com/epam/catgenome/manager/blast/BlastRequestManager.java [hunk 4]
```diff
@@ -76,11 +76,10 @@ public BlastRequestInfo getTaskStatus(final long id) throws BlastRequestExceptio
         }
     }
 
-    public BlastRequestInfo cancelTask(final long id) throws BlastRequestException {
+    public Result<BlastRequestInfo> cancelTask(final long id) throws BlastRequestException {
         try {
             Result<BlastRequestInfo> result = QueryUtils.execute(blastApi.cancelTask(id));
```
Comment (SilinPavel) [inline server/catgenome/src/main/java/com/epam/catgenome/manager/blast/BlastRequestManager.java]: could be joined

**Comment by Tatyana2022** [comment]
***Verified***

**Pull request merged**